### ISSUE TITLE
[Feature Request] Define and clean up the semantics for `REPLACE TABLE`

### ISSUE THREAD

**Issue by jackierwzhang** [delta-io/delta#1847]
### Overview

Currently the semantics for `REPLACE TABLE` is unclear, right now it acts like a combination of "emptying existing data" + "create new table". This makes it confusing with `INSERT OVERWRITE` and `.mode('overwrite').option('overwriteSchema', 'true')`. We should flesh out their semantics matrix clearly.

### Motivation

To provide better user semantics.

### Further details
N/A

### Willingness to contribute

The Delta Lake Community encourages new feature contributions. Would you or another member of your organization be willing to contribute an implementation of this feature?

- [ ] Yes. I can contribute this feature independently.
- [x] Yes. I would be willing to contribute this feature with guidance from the Delta Lake community.
- [ ] No. I cannot contribute this feature at this time.

**Comment by kailashp19** [comment]
Yes. I would be willing to contribute this feature with guidance from the Delta Lake community.

**Comment by foss-contributor** [comment]
@jackierwzhang, thank you for your contribution! This issue has been inactive for over 18 months and will be closed in one week. To keep it open, please leave a comment on this issue or contact us in the `#contributors` channel on [Delta Lake Slack](https://go.delta.io/slack). We appreciate your involvement in Delta Lake!
